src/_pytest/fixtures.py

```diff
@@ -593,6 +595,39 @@ def _get_active_fixturedef(
             raise FixtureLookupError(argname, self)
         fixturedef = fixturedefs[index]
 
+        # Check for attempted use of an async fixture by a sync test
+        # `self.scope` here is not the scope of the requested fixture, but the scope of
+        # the requester.
+        if (
+            self.scope == "function"
+            and not inspect.iscoroutinefunction(self._pyfuncitem.obj)
+            and (
+                inspect.iscoroutinefunction(fixturedef.func)
+                or inspect.isasyncgenfunction(fixturedef.func)
+            )
+        ):
+            if fixturedef._autouse:
+                warnings.warn(
+                    PytestRemovedIn9Warning(
+                        "Sync test requested an async fixture with autouse=True. "
+                        "If you intended to use the fixture you may want to make the "
+                        "test asynchronous. If you did not intend to use it you should "
+                        "restructure your test setup. "
+                        "This will turn into an error in pytest 9."
+                    ),
+                    stacklevel=3,
+                )
+            else:
+                raise FixtureLookupError(
```

Comment:
Perhaps we should not raise an error even in this case, and be conservative and instead also issue a deprecation warning?

Reply:
yeah after I realized that e.g. hypothesis will break this needs to be a deprecation